### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by ankitStackMyBiz**
CRUD with pagination

### CODE REVIEW

File: src/main/java/com/example/contactsservice/Controller/ContactController.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ContactDTO> read(@PathVariable Long id) {
+        ContactDTO contactDTO = new ContactDTO();
+        ContactEntity contactEntity = contactService.getContact(id);
+        contactDTO.setId(contactEntity.getId());
+        contactDTO.setFirstName(contactEntity.getFirstName());
+        contactDTO.setLastName(contactEntity.getLastName());
+        contactDTO.setPhone(contactEntity.getPhone());
```
Comment (RakeshMukherjee21) [inline src/main/java/com/example/contactsservice/Controller/ContactController.java:34]: You can map entity to dto at service layer

File: src/main/java/com/example/contactsservice/Controller/ContactController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/contacts")
+public class ContactController {
+
+    @Autowired
+    ContactService contactService;
```
Comment (RakeshMukherjee21) [inline src/main/java/com/example/contactsservice/Controller/ContactController.java:20]: use constructor injection

File: src/main/java/com/example/contactsservice/Service/ContactService.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ContactService {
+
+    @Autowired
+    ContactRepository contactRepository;
```
Comment (RakeshMukherjee21) [inline src/main/java/com/example/contactsservice/Service/ContactService.java:21]: use constructor injection

File: src/main/java/com/example/contactsservice/Service/ContactService.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ContactService {
```
Comment (RakeshMukherjee21) [inline src/main/java/com/example/contactsservice/Service/ContactService.java:18]: Instead of using service class directly, please create an interface with name 'ContactService' and then create a class with name 'ContactServiceImpl' and implement the interface 'ContactService'

File: src/test/java/com/example/contactsservice/Service/ContactServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        contactDTO.setFirstName("aaaaaa");
+        contactDTO.setLastName("Te");
+        contactDTO.setPhone(1234567890);
+        when(repository.getContactByPhone(anyLong())).thenReturn(null);
+        when(repository.save(any())).then(returnsFirstArg());
+        ContactEntity contactEntity = contactService.createContact(contactDTO);
+        assertNotNull(contactEntity);
+    }
+
+
```
Comment (RakeshMukherjee21) [inline src/test/java/com/example/contactsservice/Service/ContactServiceTest.java:98]: Try to avoid unnecessary space

File: src/test/java/com/example/contactsservice/Service/ContactServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    private ContactEntity getContact(ContactDTO contactDTO, ContactEntity contact) {
+        contact.setFirstName(contactDTO.getFirstName());
+        contact.setLastName(contactDTO.getLastName());
+        contact.setPhone(contactDTO.getPhone());
+        return contact;
+    }
+
+
```
Comment (RakeshMukherjee21) [inline src/test/java/com/example/contactsservice/Service/ContactServiceTest.java:200]: remove space

File: src/test/java/com/example/contactsservice/Service/ContactServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        contactDTO.setPhone(1234567890);
+
+        ContactEntity existingContact = new ContactEntity();
+        existingContact.setPhone(1234567890L);
+        when(repository.getContactByPhone(anyLong())).thenReturn(existingContact);
+        assertThrows(NotFound.class, () -> contactService.createContact(contactDTO));
+    }
+
+    @Test
+    @DisplayName("Should pass when reading contact by id")
```
Comment (RakeshMukherjee21) [inline src/test/java/com/example/contactsservice/Service/ContactServiceTest.java:140]: It covers positive test scenario, what about negative?

File: src/main/java/com/example/contactsservice/Service/ContactService.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            return allContacts.getContent();
+        } else {
+            return new ArrayList<ContactEntity>();
+        }
+    }
+
+    public ContactEntity getContact(Long id) {
+        ContactEntity contact = contactRepository.getById(id);
+
+        if (null != contact.getFirstName()) {
```
Comment (RakeshMukherjee21) [inline src/main/java/com/example/contactsservice/Service/ContactService.java:50]: Instead of checking with firstName, you can check object directly
